Userland/Libraries/LibJS/Bytecode/Generator.cpp

```diff
@@ -262,64 +262,146 @@ CodeGenerationErrorOr<void> Generator::emit_delete_reference(JS::ASTNode const&
     return {};
 }
 
-Label Generator::perform_needed_unwinds_for_labelled_break_and_return_target_block(FlyString const& break_label)
+void Generator::generate_break()
+{
+    bool last_was_finally = false;
+    // FIXME: Reduce code duplication
+    for (size_t i = m_boundaries.size(); i > 0; --i) {
+        auto boundary = m_boundaries[i - 1];
+        using enum BlockBoundaryType;
+        switch (boundary) {
+        case Break:
+            emit<Op::Jump>().set_targets(nearest_breakable_scope(), {});
+            return;
+        case Unwind:
+            if (!last_was_finally)
+                emit<Bytecode::Op::LeaveUnwindContext>();
+            last_was_finally = false;
+            break;
+        case LeaveLexicalEnvironment:
+            emit<Bytecode::Op::LeaveEnvironment>(Bytecode::Op::EnvironmentMode::Lexical);
+            break;
+        case LeaveVariableEnvironment:
```

Comment:
no real issue–just trying to get little bit more knowledge here: are lexical & variable environments handled independently? i.e. when leaving variable environment you also leave a lexical environment, so: do you need to emits here (and below)?